### PR TITLE
Add country ThreeW activities

### PR DESCRIPTION
**Pull request by puranban**
## Addresses:

- https://github.com/toggle-corp/ifrc-go-meta/issues/345
- https://github.com/toggle-corp/ifrc-go-meta/issues/346

## Changes

- Country page ongoing threeW activities 
  - Add  key figure card details section
  - Add activities map and list table
  - Add view all activities link 
  - Add csv export button

## This PR doesn't introduce:
- [x] typos
- [x] conflict markers
- [x] unwanted comments
- [x] temporary files, auto-generated files or secret keys
- [x] `console.log` meant for debugging
- [x] codegen errors

### CODE REVIEW

File: src/views/CountryOngoingActivities/index.tsx [hunk 1]
```diff
@@ -10,9 +11,26 @@ import styles from './styles.module.css';
...
+            countryId,
+            countryResponse,
+            countryResponsePending,
+        }),
+        [
+            countryId,
+            countryResponse,
+            countryResponsePending,
+        ],
+    );
```
Comment (frozenhelium) [inline src/views/CountryOngoingActivities/index.tsx]: We can just forward `outletContext` from above
```jsx
const outletContext = useOutletContext<CountryOutletContext>();

...

<Outlet context=outletContext />
```

File: src/views/CountryOngoingActivitiesThreeWActivities/i18n.json [hunk 2]
```diff
@@ -1,6 +1,28 @@
 {
     "namespace": "countryOngoingActivitiesThreeWActivities",
     "strings": {
-        "pageTitle": "Country Ongoing 3W Activities"
+        "pageTitle": "Country Ongoing 3W Activities",
```
Comment (frozenhelium) [inline src/views/CountryOngoingActivitiesThreeWActivities/i18n.json]: Let's remove unused string

File: src/views/CountryOngoingActivitiesThreeWActivities/index.tsx [hunk 3]
```diff
@@ -1,16 +1,483 @@
...
+        pageSize: ITEM_PER_PAGE,
+    });
+
+    const {
+        response: emergencyProjectListResponse,
+        pending: emergencyProjectListResponsePending,
+    } = useRequest({
+        url: '/api/v2/emergency-project/',
+        preserveResponse: true,
+        skip: (isNotDefined(countryId)),
```
Comment (frozenhelium) [inline src/views/CountryOngoingActivitiesThreeWActivities/index.tsx]:
```suggestion
        skip: isNotDefined(countryId),
```

**Review by frozenhelium — CHANGES_REQUESTED** [review]

**Review by barshathakuri — APPROVED** [review]

File: src/views/CountryOngoingActivitiesThreeWActivities/Filters/index.tsx [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+import i18n from './i18n.json';
+import styles from './styles.module.css';
+
+type DeploymentsEmergencyProjectStatus = NonNullable<GoApiResponse<'/api/v2/global-enums/'>['deployments_emergency_project_status']>[number];
+export type FilterValue = Partial<{
+    reporting_ns: number[];
+    deployed_eru: number[];
+    sector: number[];
+    status: string[];
+    country: number[];
```
Comment (samshara) [inline src/views/CountryOngoingActivitiesThreeWActivities/Filters/index.tsx]: Let's remove this as it is not used here.

File: src/views/CountryOngoingActivitiesThreeWActivities/index.tsx [hunk 5]
```diff
@@ -1,16 +1,485 @@
...
+        filtered: isFiltered,
+        limit,
+        offset,
+    } = useFilterState<FilterValue>({
+        filter: {
+            reporting_ns: [],
+            deployed_eru: [],
+            sector: [],
+            status: [],
+            country: [],
```
Comment (samshara) [inline src/views/CountryOngoingActivitiesThreeWActivities/index.tsx]: Let's remove this unused filter

**Review by samshara — APPROVED** [review]

File: src/views/CountryOngoingActivities/index.tsx [hunk 6]
```diff
@@ -1,3 +1,4 @@
+import { useMemo } from 'react';
```
Comment (frozenhelium) [inline src/views/CountryOngoingActivities/index.tsx]: `useMemo` seems to be unused here

File: src/views/CountryOngoingActivitiesThreeWActivities/Filters/index.tsx [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+                    [key]: val,
+                };
+
+                return newFilterValue;
+            });
+        }
+    }, [onChange]);
+
+    return (
+        <div className={_cs(styles.filters, className)}>
```
Comment (frozenhelium) [inline src/views/CountryOngoingActivitiesThreeWActivities/Filters/index.tsx]: Let's remove the container `div` and just use Fragment (`<>`) it from the outer container's `useGridViewInFilter` prop

File: src/views/CountryOngoingActivitiesThreeWActivities/Filters/styles.module.css [hunk 8]
```diff
@@ -0,0 +1,5 @@
+.filters {
+    display: grid;
+    grid-gap: var(--go-ui-spacing-md);
+    grid-template-columns: repeat(auto-fit, minmax(12rem, 1fr));
+}
```
Comment (frozenhelium) [inline src/views/CountryOngoingActivitiesThreeWActivities/Filters/styles.module.css]: We can use Container's filter prop for this styling

**Review by frozenhelium — CHANGES_REQUESTED** [review]

**Review by frozenhelium — APPROVED** [review]

**Pull request merged**